homeassistant/components/backup/manager.py

```diff
@@ -484,6 +502,8 @@ async def async_delete_backup(self, backup_id: str) -> dict[str, Exception]:
             if isinstance(result, BaseException):
                 raise result
 
+        self.known_backups.remove(backup_id)
```

Comment:
Should we remove the backup from known_backups always, or only if no agent returned errors?

Reply:
Probably only remove it if no errors was returned. Not completely sure though.

Reply:
OK, I changed so it's only removed if there's no error